ibtk/include/ibtk/FEDataManager.h

```diff
@@ -1255,6 +1273,15 @@ class FEDataManager : public SAMRAI::tbox::Serializable
     const InterpSpec d_default_interp_spec;
     const SpreadSpec d_default_spread_spec;
 
+    /*!
+     * after reassociating patches with elements a node may still lie
+     * outside all patches on the finest level in unusual circumstances
+     * (like when the parent integrator class does not regrid sufficiently
+     * frequently and has more than one patch level). This enum controls
+     * what we do when this problem is detected.
+     */
+    NodeOutsidePatchCheckType d_node_patch_check = NODE_OUTSIDE_ERROR;
```

Comment:
Is the default behavior in master currently `NODE_OUTSIDE_PERMIT`? If so, should we make this `NODE_OUTSIDE_WARN` instead?

Reply:
The current behavior is `NODE_OUTSIDE_PERMIT`. To be clear: here 'outside' refers to 'outside the finest patch level', not 'outside the domain'.

I'm about 60/40 in favor of raising an error in this case since I cannot imagine a circumstance where one would want the structure to be inside the domain and not interacting with the background grid, even though its not strictly backwards compatible. Is there some use case I am not aware of where we might *not* want to crash in that case?

Reply:
OK, that sounds fine. I guess I was thinking that outside the domain implies outside of the finest patch level. :-)

Reply:
Unfortunately this is not always the case and its where @rossisimone got stuck for way too long.